tests/openedx_tagging/core/tagging/test_system_defined.py

```diff
@@ -0,0 +1,236 @@
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+    tag_class_model = EmptyTestClass
+
+
+class NotIdModelObjectTag(ModelObjectTag):
+    """
+    Model ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+    tag_class_model = EmptyTestModel
+
+
+class TestUserObjectTag(UserObjectTag):
+    """
+    User ObjectTag used for testing
+    """
+
+    system_defined_taxonomy_id = 3
+
+    class Meta:
+        proxy = True
+        managed = False
+        app_label = 'oel_tagging'
+
+
+@ddt.ddt
+class TestSystemDefinedObjectTags(TestTagTaxonomyMixin, TestCase):
+    """
+    Test for generic system defined object tags
+    """
+    def test_system_defined_is_valid(self):
+        # Valid
+        assert TestUserObjectTag()._check_system_taxonomy(taxonomy=self.user_system_taxonomy)
+
+        # Null taxonomy
+        assert not UserObjectTag()._check_system_taxonomy()
+
+        # Not system defined taxonomy
+        assert not UserObjectTag()._check_system_taxonomy(taxonomy=self.taxonomy)
+
+        # Not connected with the taxonomy
+        assert not UserObjectTag()._check_system_taxonomy(taxonomy=self.user_system_taxonomy)
+
+    @ddt.data(
+        (EmptyModelObjectTag, False),  # Without associated model
+        (NotDjangoModelObjectTag, False),  # Associated class is not a Django model
+        (NotIdModelObjectTag, False),  # Associated model has not 'id' field
+        (ModelObjectTag, False), # Testing parent class validations
+        (TestUserObjectTag, True),  #Valid
+    )
+    @ddt.unpack
+    def test_model_object_is_valid(self, object_tag, assert_value):
+        args = {
+            'taxonomy': self.user_system_taxonomy,
+            'object_id': 'id',
+            'object_type': 'object',
+            'value': 'value',
+        }
+        result = object_tag(**args).is_valid(check_object=False, check_tag=False, check_taxonomy=True)
+        self.assertEqual(result, assert_value)
+
+    @ddt.data(
+        (None, True),  # Valid
+        ('user_id', False),  # Invalid user id
+        ('10000', False),  # User don't exits
+        (None, False),  # Testing parent class validations
+    )
+    @ddt.unpack
+    def test_user_object_is_valid(self, value, assert_value):
+        if assert_value:
+            user = get_user_model()(
+                username='username',
+                email='email'
+            )
+            user.save()
+            value = user.id
+
+        object_tag = TestUserObjectTag(
+            taxonomy=self.user_system_taxonomy,
+            object_id='id',
+            object_type='object',
+            value=value,
+        )
+
+        result = object_tag.is_valid(
+            check_taxonomy=True,
+            check_object=True,
+            check_tag=True,
+        )
+
+        self.assertEqual(result, assert_value)
```

Comment:
It's ok that all of these are explicitly laid out, but you can just say this if you want to:
```suggestion
        assert object_tag.is_valid() == assert_value
```

Reply:
All test has been updated with this

@pomegranited In your experience, is there a difference in using one approach or another? For my tests I have always used `self.assertEqual()`

Reply:
@ChrisChV There's no functional difference.. I just think using plain python wherever possible looks cleaner. But I'm not that fussy about it -- we're using TestClass as a base, so all of its `.assert*` methods are available regardless.